sympy/series/limits.py

```diff
@@ -263,6 +264,39 @@ def doit(self, **hints):
         if e.is_Order:
             return Order(limit(e.expr, z, z0), *e.args[1:])
 
+        if e.is_Pow:
+            if (e.has(S.Infinity) or e.has(S.NegativeInfinity)
```

Comment:
These can be combined: `if e.has(S.Infinity, S.NegativeInfinity, ...)`. That will make a single traversal of `e` instead of four.

Reply:
Yeah this can be done.